src/prefect/filesystems.py

```diff
@@ -750,23 +751,38 @@ async def get_directory(
         """
         Clones a GitHub project specified in `from_path` to the provided `local_path`; defaults to cloning
         the repository reference configured on the Block to the present working directory.
+
+        Args:
+            - from_path: if provided, interpreted as a subdirectory of the underlying repository that will
```

Comment:
Testing the `from_path` argument is tricky without cloning a repo with every test run; will need to think about this more.

Reply:
In less time than it would take to clone a reference repo, we could `git init` and `git commit` what we wanted into a `tmp_path`.  Should be pretty easy with pytest fixtures:


```
from functools import partial
from pathlib import Path
from subprocess import check_call

@pytest.fixture
def repo(tmp_path: Path) -> Path:
    in_repo = partial(check_call, cwd=tmp_path)

    in_repo(['git', 'init', '-b', 'main'])

    (tmp_path / "my_flow.py").write_text(...some flow code...)

    in_repo(['git', 'add', '.'])
    in_repo(['git', 'commit', '--message', 'initial commit')

    return tmp_path

@pytest.fixture
def prod_branch(repo: Path) -> Path:
    in_repo = partial(check_call, cwd=tmp_path)

    in_repo(['git', 'checkout', '-b', 'prod'])

    return repo
```